urllib3/sessionmanager.py

```diff
@@ -0,0 +1,77 @@
+from .util.retry import Retry
+from .request import RequestMethods, Request
+from .util.sessioncontext import SessionContext
+
+
+class SessionManager(RequestMethods):
+    """
+    Allows arbitrary requests while maintaining session context across
+    those requests. Currently, that context consists of automatic
+    cookie storage and retrieval.
+
+    :param manager:
+        An appropriate :class:`urllib3.poolmanager.PoolManager` or
+        :class:`urllib3.poolmanager.ProxyManager` object
+        to handle HTTP requests for the SessionManager
+
+    :param context:
+        A predefined :class:`urllib3.util.context.SessionContext` object to use in the session;
+        if not provided, a new one will be created.
+
+    :param headers:
+        A set of headers to include with all requests, unless other
+        headers are given explicitly.
```

Comment:
Maybe avoid the use of the word `set` here? It's also the name of a data type, and so this can confuse users who aren't thinking carefully about how this might work.